.github/workflows/deploy.py

```diff
@@ -0,0 +1,150 @@
+IMAGE_NAME = 'neonlabsorg/evm_loader'
+
+client = docker.from_env()
+api_client = docker.APIClient()
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command(name="build_docker_image")
+@click.option('--github_sha')
+def build_docker_image(github_sha):
+    solana_revision = 'v1.11.10'
+    solana_image = f'solanalabs/solana:{solana_revision}'
+    client.images.pull(solana_image)
+    buildargs = {"REVISION": github_sha,
+                 "SOLANA_IMAGE": solana_image,
+                 "SOLANA_REVISION": solana_revision}
+
+    tag = f"{IMAGE_NAME}:{github_sha}"
+    click.echo("start build")
+    output = api_client.build(tag=tag, buildargs=buildargs, path="./")
+
+    for line in output:
+        if 'stream' in str(line):
+            click.echo(str(line).strip('\n'))
+
+
+@cli.command(name="publish_image")
+@click.option('--branch')
+@click.option('--github_sha')
+def publish_image(branch, github_sha):
+    if branch == 'master':
+        tag = 'stable'
+    elif branch == 'develop':
+        tag = 'latest'
+    else:
+        tag = branch.split('/')[-1]
+
+    image = client.images.get(f"{IMAGE_NAME}:{github_sha}")
+    image.tag(f"{IMAGE_NAME}:{github_sha}", tag)
+    client.login(username=DOCKER_USER, password=DOCKER_PASSWORD)
+    client.images.push(f"{IMAGE_NAME}:{tag}")
+
+
+@cli.command(name="run_tests")
+@click.option('--github_sha')
+def run_tests(github_sha):
+    image_name = f"{IMAGE_NAME}:{github_sha}"
+
+    os.environ["EVM_LOADER_IMAGE"] = image_name
+
+    command = "docker-compose -f ./evm_loader/docker-compose-test.yml down --timeout 1"
+    click.echo(f"run command: {command}")
+    subprocess.run(command, shell=True)
+
+    command = "docker-compose -f ./evm_loader/docker-compose-test.yml up -d"
+    click.echo(f"run command: {command}")
+
+    subprocess.run(command, stdout=subprocess.PIPE, shell=True)
+
+    try:
+        click.echo("start tests")
+        container = client.containers.get("solana")
+        logs = container.exec_run(cmd="/opt/deploy-test.sh")
+        click.echo(logs)
+    except:
+        raise "Solana container is not run"
+
+
+@cli.command(name="trigger_proxy_action")
+@click.option('--branch')
+@click.option('--github_sha')
+@click.option('--token')
+@click.option('--is_draft')
+def trigger_proxy_action(branch, github_sha, token, is_draft):
+    proxy_endpoint = "https://api.github.com/repos/neonlabsorg/proxy-model.py"
+    proxy_endpoint = "https://api.github.com/repos/kristinaNikolaeva/playwright_autotests"
+    proxy_branches_obj = requests.get(f"{proxy_endpoint}/branches").json()
+    proxy_branches = [item["name"] for item in proxy_branches_obj]
+    if branch not in proxy_branches:
+        branch = 'develop'
+
+    if branch == "develop" and not is_draft:
+        full_test_suite = True
+    else:
+        full_test_suite = False
+
+    data = {"ref": f"refs/heads/{branch}",
+            "inputs": {"full_test_suite": full_test_suite}}
+    headers = {'Authorization': f'Bearer {token}',
```

Comment:
github provide api token in  GITHUB_TOKEN but I'm not sure that it work for all repos

Reply:
Do you mean enviroment variables?

Reply:
https://docs.github.com/en/actions/security-guides/automatic-token-authentication